streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -655,21 +622,32 @@ public final <KIn, VIn> void addGlobalStore(final String sourceName,
                                                 final String processorName,
                                                 final ProcessorSupplier<KIn, VIn, Void, Void> stateUpdateSupplier,
                                                 final boolean reprocessOnRestore) {
-        ApiUtils.checkSupplier(stateUpdateSupplier);
+        verifyName(sourceName);
+
+        Objects.requireNonNull(topic, "topic cannot be null");
+        validateTopicNotAlreadyRegistered(topic);
+
+        verifyName(processorName);
+        if (sourceName.equals(processorName)) {
+            throw new TopologyException("sourceName and processorName must be different.");
+        }
+
         final Set<StoreBuilder<?>> stores = stateUpdateSupplier.stores();
         if (stores == null || stores.size() != 1) {
             throw new IllegalArgumentException(
-                    "Global stores must pass in suppliers with exactly one store but got " +
-                            (stores != null ? stores.size() : 0));
+                "Global stores must pass in suppliers with exactly one store but got " +
+                    (stores != null ? stores.size() : 0));
         }
         final StoreFactory storeFactory =
-                StoreBuilderWrapper.wrapStoreBuilder(stores.iterator().next());
-        validateGlobalStoreArguments(sourceName,
```

Comment:
This method is only called once. Removed it an pulled in the code. (Allow to use the new helper method better.)